volatility3/framework/interfaces/layers.py

```diff
@@ -551,6 +597,12 @@ def _scan_iterator(
         assumed to have no holes
         """
         for section_start, section_length in sections:
+            # Check the virtmap cache and use it if available
```

Comment:
But, isn't that exactly what this *is* generating here?  Why not just save it rather than generating it in a different plugin as part of a separate step?  This could also build up the cache as needed, so if parts haven't been computed yet, it does the full task, but if they are it uses the cache?

Implementing a cache (that might be wrong unless it's matched to the image very carefully) already seems like quite a deep-rooted feature to me...